File: Number Operation/prg.java
import java.util.Scanner;

public class LogicalBinaryOps {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt(), b = sc.nextInt();

        System.out.println("AND: " + (a & b));
        System.out.println("OR: " + (a | b));
        System.out.println("XOR: " + (a ^ b));
        System.out.println("NOT A: " + (~a));
        System.out.println("Left Shift: " + (a << 1));
        System.out.println("Right Shift: " + (a >> 1));

        boolean x = sc.nextBoolean(), y = sc.nextBoolean();
        System.out.println("Logical AND: " + (x && y));
        System.out.println("Logical OR: " + (x || y));
        System.out.println("Logical NOT X: " + (!x));
        
        sc.close();
    }
}

File: Number Operation/prg.kt
fun main() {
    val a = readLine()!!.toInt()
    val b = readLine()!!.toInt()

    println("AND: ${a and b}")
    println("OR: ${a or b}")
    println("XOR: ${a xor b}")
    println("NOT A: ${a.inv()}")
    println("Left Shift: ${a shl 1}")
    println("Right Shift: ${a shr 1}")

    val x = readLine()!!.toBoolean()
    val y = readLine()!!.toBoolean()
    println("Logical AND: ${x && y}")
    println("Logical OR: ${x || y}")
    println("Logical NOT X: ${!x}")
}
